### ISSUE TITLE
TF2.0: Non-Linear Model Produces Linear Outputs

### ISSUE THREAD

**Issue by James-Montgomery** [tensorflow/tensorflow#28292]
**System information**
- Included is a minimal viable example of code to produce this error
- OS Platform and Distribution: MacOS Mojave 10.14.4
- TensorFlow installed from pypi: `pip install -q tensorflow==2.0.0-alpha0`
- TensorFlow version: v1.12.0-9492-g2c319fb415 2.0.0-alpha0
- Python version: 3.6.8

**Current Behavior**
I've been having a lot of trouble using tf2 to make sensible predictions for regression use cases. When I specify a model that should provide non-linear outputs, the model still provides linear outputs. I've tried a number of different ways of specifying the model such as using the keras sequential model api and specifying my own model using the keras subclassing api. I get consistent results on each. **I've shown a minimal viable example of the issue I am seeing below. **

**Expected Behavior**
If specifying a non-linear model the predictions should be non-linear. For example, in the minimal viable example I provide, the model specified has 3 hidden layers with 10 units each and relu activation functions and yet produces a linear output without training. The output should, instead, be some linear function whose flexibility and smoothness is determined by the number fo hidden layers and the chosen activation function.

**Code to reproduce the issue**
~~~Python
import tensorflow as tf
from tensorflow.keras.layers import Dense, Dropout
from tensorflow.keras import Model

import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

N = 100
x = np.random.uniform(0, 10, N)
y = 0.5*x + np.random.normal(0, 2.5, N)

x = x.reshape(-1, 1)
y = y.reshape(-1, 1)

model = tf.keras.models.Sequential([
  tf.keras.layers.Dense(10, activation='relu'),
  tf.keras.layers.Dense(10, activation='relu'),
  tf.keras.layers.Dense(10, activation='relu'),
  tf.keras.layers.Dense(1)
])

model.compile(optimizer='adam',
              loss='mean_squared_error')

#model.fit(x, y, epochs=50, verbose=0)

predictions = model(x).numpy()
print(predictions.shape)

plt.scatter(x, y)
plt.scatter(x, predictions)
plt.show()
~~~

**Other info / logs**
![screencapture-localhost-8888-notebooks-Untitled1-ipynb-2019-04-30-15_14_04](https://user-images.githubusercontent.com/21271116/56988332-c30c3b00-6b5d-11e9-8d59-2e1cae3bca23.png)

**Comment by James-Montgomery** [comment]
This is not an issue with tensorflow, but rather a model misspecification.
